apis/clusters/v1beta1/structs.go

```diff
@@ -820,3 +820,110 @@ func (g GenericResizeSettings) Equal(o GenericResizeSettings) bool {
 
 	return true
 }
+
+type AWSSettings struct {
+	EncryptionKey          string `json:"encryptionKey,omitempty"`
+	CustomVirtualNetworkID string `json:"customVirtualNetworkID,omitempty"`
+	BackupBucket           string `json:"backupBucket,omitempty"`
+}
+
+type GCPSettings struct {
+	CustomVirtualNetworkID    string `json:"customVirtualNetworkID,omitempty"`
+	DisableSnapshotAutoExpiry string `json:"disableSnapshotAutoExpiry,omitempty"`
+}
+
+type AzureSettings struct {
+	ResourceGroup          string `json:"resourceGroup,omitempty"`
+	CustomVirtualNetworkID string `json:"customVirtualNetworkID,omitempty"`
+	StorageNetwork         string `json:"storageNetwork,omitempty"`
+}
+
+type CloudProviderOpts struct {
+	AWSSettings   *AWSSettings   `json:"awsSettings,omitempty"`
+	GCPSettings   *GCPSettings   `json:"gcpSettings,omitempty"`
+	AzureSettings *AzureSettings `json:"azureSettings,omitempty"`
+}
+
+type CloudProviderOptions []*CloudProviderOpts
+
+func (c *CloudProviderOptions) FromInstAPI(instaModel *models.CloudProviderSettings) {
+	s := CloudProviderOpts{}
+	*c = append(*c, &s)
+
+	switch {
+	case len(instaModel.AWSSettings) > 0:
+		s.AWSSettings = &AWSSettings{
+			EncryptionKey:          instaModel.AWSSettings[0].EBSEncryptionKey,
+			CustomVirtualNetworkID: instaModel.AWSSettings[0].CustomVirtualNetworkID,
+			BackupBucket:           instaModel.AWSSettings[0].BackupBucket,
+		}
+	case len(instaModel.GCPSettings) > 0:
+		s.GCPSettings = &GCPSettings{
+			CustomVirtualNetworkID:    instaModel.GCPSettings[0].CustomVirtualNetworkID,
+			DisableSnapshotAutoExpiry: instaModel.GCPSettings[0].DisableSnapshotAutoExpiry,
+		}
+	case len(instaModel.AzureSettings) > 0:
+		s.AzureSettings = &AzureSettings{
+			ResourceGroup:          instaModel.AzureSettings[0].ResourceGroup,
+			CustomVirtualNetworkID: instaModel.AzureSettings[0].CustomVirtualNetworkID,
+			StorageNetwork:         instaModel.AzureSettings[0].StorageNetwork,
+		}
+	}
+}
```

Comment:
Here we will append only if `s` is not nil:
```
func (c *CloudProviderOptions) FromInstAPI(instaModel *models.CloudProviderSettings) {
	var s *CloudProviderOpts

	switch {
	case len(instaModel.AWSSettings) > 0:
		s = &CloudProviderOpts{
			AWSSettings: &AWSSettings{
				EncryptionKey:          instaModel.AWSSettings[0].EBSEncryptionKey,
				CustomVirtualNetworkID: instaModel.AWSSettings[0].CustomVirtualNetworkID,
				BackupBucket:           instaModel.AWSSettings[0].BackupBucket,
			},
		}
	case len(instaModel.GCPSettings) > 0:
		s = &CloudProviderOpts{
			GCPSettings: &GCPSettings{
				CustomVirtualNetworkID:    instaModel.GCPSettings[0].CustomVirtualNetworkID,
				DisableSnapshotAutoExpiry: instaModel.GCPSettings[0].DisableSnapshotAutoExpiry,
			},
		}
	case len(instaModel.AzureSettings) > 0:
		s = &CloudProviderOpts{
			AzureSettings: &AzureSettings{
				ResourceGroup:          instaModel.AzureSettings[0].ResourceGroup,
				CustomVirtualNetworkID: instaModel.AzureSettings[0].CustomVirtualNetworkID,
				StorageNetwork:         instaModel.AzureSettings[0].StorageNetwork,
			},
		}

		if s != nil {
			*c = append(*c, s)
		}
	}
}
```

wdyt?

Reply:
It makes sense. Done.